sklearn/utils/testing.py

```diff
@@ -697,5 +706,34 @@ def check_skip_travis():
     if os.environ.get('TRAVIS') == "true":
         raise SkipTest("This test needs to be skipped on Travis")
 
+
+def delete_folder(folder_path):
+    """Utility function to cleanup a temporary folder if still existing. Copy from joblib.pool (for independance)"""
+    if os.path.exists(folder_path):
+        shutil.rmtree(folder_path)
```

Comment:
You should still put this call under a `try / except shutil.WindowsError:` block and add `warn=False` argument to `delete_folder`.

`warn` should be set to `True` only when called via `atexit`. When called via `__exit__` we should not warn.